lib/dataset/test/data_array_arithmetic_test.cpp

```diff
@@ -11,16 +11,118 @@
+  EXPECT_EQ(res1.coords()[Dim::X], coord_1);
+  EXPECT_TRUE(res1.coords()[Dim::X].is_aligned());
+
+  const auto res2 = da_2 + da_1;
+  EXPECT_EQ(res2.coords()[Dim::X], coord_1);
+  EXPECT_TRUE(res2.coords()[Dim::X].is_aligned());
+}
+
+TEST(DataArrayArithmeticTest, merge_coords_alignment) {
+  const auto coord_1 =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 3});
+  auto coord_2 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  auto coord_3 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  auto coord_4 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  auto coord_5 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  const auto data =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+
+  coord_3.set_aligned(false);
+  coord_5.set_aligned(false);
+  const DataArray da_1(data, {{Dim::X, coord_1},
+                              {Dim::Y, copy(coord_2)},
+                              {Dim::Z, copy(coord_3)},
+                              {Dim{"1.4"}, coord_4},
+                              {Dim{"1.5"}, coord_5}});
+
+  coord_2.set_aligned(false);
+  coord_3.set_aligned(true);
+  const DataArray da_2(data, {{Dim::X, coord_1},
+                              {Dim::Y, coord_2},
+                              {Dim::Z, coord_3},
+                              {Dim{"2.4"}, coord_4},
+                              {Dim{"2.5"}, coord_5}});
+
+  const auto res = da_1 + da_2;
+  EXPECT_TRUE(res.coords()[Dim::X].is_aligned());
+  EXPECT_TRUE(res.coords()[Dim::Y].is_aligned());
+  EXPECT_TRUE(res.coords()[Dim::Z].is_aligned());
+  EXPECT_TRUE(res.coords()[Dim{"1.4"}].is_aligned());
+  EXPECT_FALSE(res.coords()[Dim{"1.5"}].is_aligned());
+  EXPECT_TRUE(res.coords()[Dim{"2.4"}].is_aligned());
+  EXPECT_FALSE(res.coords()[Dim{"2.5"}].is_aligned());
+}
+
+TEST(DataArrayArithmeticTest, operation_does_not_overwrite_input_alignment) {
+  const auto coord_1 =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 3});
+  auto coord_2 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  auto coord_3 = makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+  const auto data =
+      makeVariable<double>(Dims{Dim::X}, Shape{3}, Values{1, 2, 4});
+
+  coord_3.set_aligned(false);
+  DataArray da_1(
+      data,
+      {{Dim::X, coord_1}, {Dim::Y, copy(coord_2)}, {Dim::Z, copy(coord_3)}});
+
+  coord_2.set_aligned(false);
+  coord_3.set_aligned(true);
+  DataArray da_2(data,
+                 {{Dim::X, coord_1}, {Dim::Y, coord_2}, {Dim::Z, coord_3}});
+
+  [[maybe_unused]] const auto res = da_1 + da_2;
+  EXPECT_TRUE(da_1.coords()[Dim::X].is_aligned());
+  EXPECT_TRUE(da_1.coords()[Dim::Y].is_aligned());
+  EXPECT_FALSE(da_1.coords()[Dim::Z].is_aligned());
+  EXPECT_TRUE(da_2.coords()[Dim::X].is_aligned());
+  EXPECT_FALSE(da_2.coords()[Dim::Y].is_aligned());
+  EXPECT_TRUE(da_2.coords()[Dim::Z].is_aligned());
+}
+
```

Comment:
I can't see any new tests for inplace ops such as `+=`. Are those unaffected?

Reply:
Did some manual testing. Currently, this behaves almost like attrs. That is, the LHS coord is kept regardless of alignment flags. But if a coord is missing in LHS, we get an exception. If it is missing in the RHS, it is ignored.

Is this what we want? Seems that, at least, if the coord is missing in the LHS and unaligned in the RHS, the op goes through and the result has no coord.

Reply:
> Seems that, at least, if the coord is missing in the LHS and unaligned in the RHS, the op goes through and the result has no coord.

That makes sense.

What if RHS has aligned coord?

Reply:
> What if RHS has aligned coord?

Exception if LHS doesn't have it or it differs.

Reply:
Is this unchanged compared to know?